backend/capellacollab/projects/toolmodels/diagrams/routes.py

```diff
@@ -67,14 +67,28 @@ async def get_diagram_metadata(
     )
 
 
-@router.get("/{diagram_uuid}", response_class=fastapi.responses.Response)
+@router.get(
+    "/{diagram_uuid_or_filename}", response_class=fastapi.responses.Response
+)
 async def get_diagram(
-    diagram_uuid: str,
+    diagram_uuid_or_filename: str,
     handler: git_handler.GitHandler = fastapi.Depends(
         git_injectables.get_git_handler
     ),
     logger: logging.LoggerAdapter = fastapi.Depends(log.get_request_logger),
 ):
+    splitted = diagram_uuid_or_filename.split(".")
+    if len(splitted) > 1:
+        diagram_uuid, fileextension = splitted
```

Comment:
Unpacking will fail if `len(splitted) > 2`. We have to consider this case. We can use pathlib to parse the file name and to find out the suffix: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffix

Reply:
I have seen this with the Path suffix - alternatively we could also use rsplit again https://github.com/DSD-DBS/capella-collab-manager/pull/1090#discussion_r1356658664

Reply:
I'd prefer the PurePosixPath suffix.